src/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -91,6 +100,18 @@ public override void Write(byte[] buffer, int offset, int count)
             _buffer.Write(buffer, offset, count);
         }
 
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            Write(buffer, offset, count);
+            
+            return Task.CompletedTask;
+        }
```

Comment:
The Write overload is implemented as:
_buffer.Write(buffer, offset, count);

For parity and simplicity, you could implement this just as:
return _buffer.WriteAsync(buffer, offset, count, cancellationToken);

MemoryStream implements WriteAsync appropriately.